### PR TITLE
8257565: epsilonBarrierSet.hpp should not include barrierSetAssembler

### PR DESCRIPTION
**Pull request by iklam**
Please review this trivial fix:

epsilonBarrierSet.hpp is included (recursively via access.hpp) by many CPP files. It unncessarily includes of barrierSetAssembler.hpp, which causes many of the native code assembler header files to be unnecessarily included by many HotSpot CPP files that do not deal with native code assembly.

Removing this one line reduced the total number of header inclusion for building HotSpot from 260096 to 258193, or about 0.8%.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257565](https://bugs.openjdk.java.net/browse/JDK-8257565): epsilonBarrierSet.hpp should not include barrierSetAssembler

### Reviewers
 * [Kim Barrett](https://openjdk.java.net/census#kbarrett) (@kimbarrett - **Reviewer**) ⚠️ Review applies to 3949a648f7e133f0159ee674fe06af9ccacb6c03
 * [Thomas Stuefe](https://openjdk.java.net/census#stuefe) (@tstuefe - **Reviewer**) ⚠️ Review applies to 3949a648f7e133f0159ee674fe06af9ccacb6c03
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 82b910855e9d25fe55a0ed2beda08d01bf67c6dc

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1554/head:pull/1554`
`$ git checkout pull/1554`

### CODE REVIEW

**Review by kimbarrett — APPROVED** [review]

**Comment by tstuefe** [comment]
LGTM.

**Review by tstuefe — APPROVED** [review]

File: src/hotspot/share/gc/epsilon/epsilonBarrierSet.hpp [hunk 1]
```diff
@@ -1,4 +1,5 @@
 /*
+ * Copyright (c) 2020, Oracle and/or its affiliates. All rights reserved.
```
Comment (shipilev) [inline src/hotspot/share/gc/epsilon/epsilonBarrierSet.hpp:2]: Why the copyright line addition, though? It is not like Red Hat adds its copyrights for trivial changes to shared files like these.

**Review by shipilev — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/epsilon/epsilonBarrierSet.hpp (hunk 1 above)
Comment (iklam) [reply src/hotspot/share/gc/epsilon/epsilonBarrierSet.hpp]: Removed.

**Review by shipilev — APPROVED** [review]
Looks good! Thanks, I'll take care of backports, if any.

**Comment by iklam** [comment]
@shipilev @tstuefe @kimbarrett Thanks for the review. I've tested tier1,builds-tier2,builds-tier3,builds-tier4,builds-tier5 on mach5, plus locally building s390x,ppc64le.
/integrate

**Pull request closed without merging**